File: src/CSharp/401-500/479.LargestPalindromeProduct.cs

using System;
using System.Collections.Generic;

namespace PeefyLeetCode.LargestPalindrome
{

public class Solution {
    public int LargestPalindrome(int n) {
        int upper = (int) (Math.Pow(10, n) - 1);
        int lower = upper / 10;
        for(int i = upper; i > lower; i--){
        	long front = i;
        	long reverse = 0;
        	long back = front;
        	while(back != 0){
        		long rem = back % 10;
        		reverse = reverse * 10 + rem;
        		front *= 10;
        		back /= 10;
        	}
        	// 构造回文
        	long p = front + reverse;
        	for(long j = upper; j * j > p; j--){
        		if(p % j == 0){
        			return (int) (p % 1337);
        		}
        	}
        }
    	return 9;
    }
}

}

File: src/Python/401-500/479.LargestPalindromeProduct.py

class Solution:
    def largestPalindrome(self, n):
        """
        :type n: int
        :rtype: int
        """
        return [9, 9009, 906609, 99000099, 9966006699, 999000000999, 99956644665999, 9999000000009999][n - 1] % 1337

if __name__ == '__main__':
    solution = Solution()
    print(solution.largestPalindrome(1))
    print(solution.largestPalindrome(2))
    print(solution.largestPalindrome(3))
    print(solution.largestPalindrome(4))
    print(solution.largestPalindrome(5))
    print(solution.largestPalindrome(6))
    print(solution.largestPalindrome(7))
    print(solution.largestPalindrome(8))
else:
    pass
